components/cloud-foundry/frontend/src/view/applications/application/services/services.html

```diff
@@ -32,11 +21,32 @@ <h3 translate>There are no services available for this endpoint. Please contact
     </form>
   </div>
 </div>
-
-<div class="panel panel-default"
-  ng-repeat="service in applicationServicesCtrl.services | filter:applicationServicesCtrl.category:true | filter:applicationServicesCtrl.search track by service.metadata.guid">
-  <div class="panel-body service-panel">
-    <service-card service="service" app="applicationServicesCtrl.appModel.application"
-      cnsi-guid="applicationServicesCtrl.cnsiGuid"></service-card>
-  </div>
+<div class="service-manage-message" ng-if="applicationServicesCtrl.managingType">You are currently viewing <b>{{::applicationServicesCtrl.managingType}}</b> instances. <a ng-click="applicationServicesCtrl.filterType = applicationServicesCtrl.ALL_FILTER">Click here</a> to view all types.</div>
+<div ng-if="applicationServicesCtrl.ready && boundServices.length === 0"
+  class="message-box message-box-no-bg">
+  <p>
+    <img src="/images/services/App_services_72.png" />
+    <div ng-if="!applicationServicesCtrl.search.$ && !applicationServicesCtrl.isFiltered()">
+      <h3>
+        This application has no bound service instances.
+      </h3>
+      <h3>
+        To view the avaliable services for this application, please visit the <a ui-sref="cf.applications.application.service-catalogue">service catalogue</a>.
+      </h3>
+    </div>
+    <div ng-if="applicationServicesCtrl.isFiltered()">
+      <h3 translate>
+        No services instances found matching your search.
+      </h3>
+    </div>
+  </p>
+</div>
+<div ng-if="!applicationServicesCtrl.ready" class="message-box message-box-no-bg">
+  <bounce-spinner classes="bounce-spinner-sm"></bounce-spinner>
+</div>
+<div class="cluster-gallery-card-container">
+  <service-card class="cluster-gallery-card-parent" ng-repeat="service in boundServices = (applicationServicesCtrl.services | filter:applicationServicesCtrl.showInstance | filter:applicationServicesCtrl.search) track by service.metadata.guid"
```

Comment:
The first and last service cards have respectively left and right padding. We don't show these on other card layouts. Squishing and expanding all works fine.